torch/_meta_registrations.py

```diff
@@ -2655,6 +2654,37 @@ def meta_upsample_bilinear2d_aa(
     )
 
 
+# From aten/src/ATen/native/cuda/AmpKernels.cu
+@register_meta(aten._amp_foreach_non_finite_check_and_unscale_.default)
+def _amp_foreach_non_finite_check_and_unscale_(self, found_inf, inv_scale):
+    check(found_inf.numel() == 1, lambda: "found_inf must be a 1-element tensor.")
+    check(inv_scale.numel() == 1, lambda: "inv_scale must be a 1-element tensor.")
+    check(
+        found_inf.dtype.is_floating_point, lambda: "found_inf must be a float tensor."
+    )
+    check(
+        inv_scale.dtype.is_floating_point, lambda: "inv_scale must be a float tensor."
+    )
+
+
+# From aten/src/ATen/native/UnaryOps.cpp
+@register_meta(aten.nan_to_num.default)
+def nan_to_num(self, nan=None, posinf=None, neginf=None):
+    result_size = list(self.size())
+    return self.new_empty(result_size)
+
+
+# From aten/src/ATen/native/UnaryOps.cpp
+@register_meta(aten.nan_to_num.out)
```

Comment:
nit - you should be able to just add `@out_wrapper` on top of the `def nan_to_num` implementation, to get an out= meta for free. Can you try that?

Can you also add a small unit test for nan_to_num, inside of `test/test_meta.py`? I dont think we have any tests for it.

Reply:
Updated with `@out_wrapper` and added a small meta unit test for `nan_to_num`:
```
    def test_nan_to_num(self):
        t = torch.tensor([float('nan'), float('inf'), -float('inf'), 3.14], device='meta')
        r = x.nan_to_num()
        self.assertEqual(r.device.type, 'meta')
```